uaclient/files/state_files.py

```diff
@@ -295,3 +295,12 @@ def __init__(self, attached_at: datetime.datetime):
     defaults.DEFAULT_PRIVATE_DATA_DIR,
     private=True,
 )
+
+
+def delete_state_files():
+    machine_id_file.delete()
+    status_cache_file.delete()
+    attachment_data_file.delete()
+    anbox_cloud_credentials_file.delete()
+    reboot_cmd_marker_file.delete()
+    services_once_enabled_file.delete()
```

Comment:
Should the `user_config_file` also be deleted with the state files here?

Reply:
No because it holds config value that are relevant while attached and this function is called on detach.

Maybe this function should be renamed to "delete_state_files_on_detach" - or does it need to be a function at all? Should these delete calls just go into the detach function (next to `machine_token_file.delete()`)?

Reply:
To be honest, I did that to make mocking in the unittest simpler. or we would need to individually mock all of the files for the happy path of the detach. But I can rename the function if you prefer